### PR TITLE
[MOB-9745] Update mobile SDKs to support fetching a new JWT to use for the destinationUserId / destinationEmail when calling merge

### PR DESCRIPTION
**Pull request by megha-iterable**
## Jira Ticket(s) if any

* [MOB-9745](https://iterable.atlassian.net/browse/MOB-9745)

## Description

> Update mobile SDKs to support fetching a new JWT to use for the destinationUserId / destinationEmail when calling merge

[MOB-9745]: https://iterable.atlassian.net/browse/MOB-9745?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableAuthManager.java [hunk 1]
```diff
@@ -39,8 +39,8 @@ public class IterableAuthManager {
         this.expiringAuthTokenRefreshPeriod = expiringAuthTokenRefreshPeriod;
     }
 
-    public synchronized void requestNewAuthToken(boolean hasFailedPriorAuth) {
-        requestNewAuthToken(hasFailedPriorAuth, null, true);
+    public synchronized void requestNewAuthToken(boolean hasFailedPriorAuth, GenerateJWTSuccess generateJwtSuccess) {
```
Comment (Ayyanchira) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableAuthManager.java:42]: what does GenerateJWTSuccess mean in terms of JWT workflow?

Comment (evantk91) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableAuthManager.java:42]: I think this is providing a callback to allow the customers backend to pass in a JWT for the anonymous user id

Comment (megha-iterable) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableAuthManager.java:42]: Yes, It's callback for JWT generate successfully and then after we can call the sync/merge events with latest JWT (with destination userId/email)

**Review by evantk91 — APPROVED** [review]

**Comment by evantk91** [comment]
Making changes to this PR per our discussion
* removing GenerateJWTSuccess callback in favor of using success callback already implemented
* creating a new function attemptMergeAndEventReplay that takes in email/userI
* passing in email/userId into onLogin function to call attemptMergeAndEventReplay within onLogin function

File: app/src/main/java/com/iterable/androidsdk/AnonTrackingTestActivity.java [hunk 2]
```diff
@@ -150,4 +158,48 @@ public void printAllSharedPreferencesData(Context context) {
     public void onAnonUserCreated(String userId) {
         Log.d("userId", userId);
     }
+
+    @Override
+    public String onAuthTokenRequested() {
+        IterableApi instance = IterableApi.getInstance();
+        if (instance.getAuthToken() == null) {
+            final String secret = "1bb125ddcda2808f118c8b5e774d341c4b03fae68ebef0d140a22da1ec0295ad24d98981fd262c93bac98fa2e63a08142c0a36fe4322c09bea90f48c161780e0";
```
Comment (joaodordio) [inline app/src/main/java/com/iterable/androidsdk/AnonTrackingTestActivity.java:166]: This secret key should not be hardcoded into the repo. Can this be moved into a environment variable that's read from a local.properties file instead?

Comment (joaodordio) [reply app/src/main/java/com/iterable/androidsdk/AnonTrackingTestActivity.java:166]: Never mind, just noticed this is on the Test Activity 🤦

**Review by evantk91 — APPROVED** [review]

**Pull request merged**